htdocs/product/class/html.formproduct.class.php

```diff
@@ -865,6 +865,25 @@ private function loadLotStock($productIdArray = array())
 			// clear cache
 			$this->cache_lot = array();
 			$productIdList = implode(',', $productIdArray);
+
+			$batch_count = 0;
+			include_once DOL_DOCUMENT_ROOT . '/core/class/hookmanager.class.php';
```

Comment:
To dangerous because if this function is called in a hook you lost the previous result context and cause a hazard effect
This happend time to time ...
Debuging this case take time and it's not found immediatly ...

Reply:
I don't understand. This is the reason why the hookmanager must not be create anywhere but only at the top of the page. And then all the code must use all the same $hookmanager. Creating another class factory erase the old one and you loose the context and the init already done.

Reply:
It's not this problem. The case is a hook who is called when you are already in a hook.
But a can make as you wish.

Reply:
Even if a hook is inside a hook, the hookmanager should be the same and unique for all the "page session", as it is a global object (like $conf, $db, $langs, $user)
But this show this rule is not apply everywhere in Dolibarr code. i will make a clean of Dolibarr code so hook will be created only in parent top page. We should not create new one anymore inside low level code.